### DISCUSSION TITLE
Is there any point of the anonymous wrapper function as of now?

### DISCUSSION THREAD

**Discussion by Destroy666x** [Tampermonkey/tampermonkey#1978] [Q&A] [answered]
From what I understand user scripts are wrapped anyways. Or am I mistaken?

So why is:
```
(function() {
})();
```
added to new scripts automatically?

**Comment by derjanb** [comment, marked as answer]
Because of the `'use strict';` statement inside..

**Reply by Destroy666x** [reply]
That's it? Why not add some kind of optional `@strict-mode` param to the script definition in that case? Would improve the readability IMO.

**Reply by 7nik** [reply]
Adding `@require` will make the root `'use strict';` not the first statement in the script.

Edit: adding `'use strict';` before the required scripts will cause running them in the strict mode, though they may be incompatible with it.

**Reply by Destroy666x** [reply]
I see, so the requires are part of the wrapper? Thanks for info @7nik

**Reply by 7nik** [reply]
The required scripts are inserted inside the wrapper before your code, like if you copy-pasted them to the beginning of the script.
